Cargo.toml

```diff
@@ -17,11 +17,10 @@ async-std-runtime = ["async-std", "async-std/attributes"]
 
 [dependencies]
```

Comment:
One change made as part of this PR is to migrate away from `tokio::io` based dependencies in favor of the `futures::io` ones. While the tokio ones are fine for our needs, it appears that the community (including `async-std` itself) is adopting the `futures::io` ones more often for runtime-agnostic contexts. (e.g. `async-tls`, `trust-dns`). 

The `AsyncWrite` / `AsyncRead` traits are nearly identical between `futures` and `tokio` (`poll_shutdown` vs `poll_close` being the main difference), so it required very little changes on our side though. `AsyncWriteExt` and `AsyncReadExt` are more fleshed out in tokio though, so I had to supplement them in the driver.